lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1271,33 +1295,51 @@ def _on_logging_relation_departed(self, event: RelationDepartedEvent):
             event: a `CharmEvent` in response to which the Loki
                 charm must update its relation data.
         """
-        self._update_alert_rules(event.relation)
-        self._check_alert_rules()
+        self.on.loki_push_api_alert_rules_changed.emit(
+            relation=event.relation,
+            relation_id=event.relation.id,
+            app=self._charm.app,
+            unit=self._charm.unit,
+        )
 
-    def _process_logging_relation_changed(self, relation: Relation):
+    def _should_update_alert_rules(self, relation) -> bool:
```

Comment:
The function name is a bit misleading: it will continue to return True even after alert rules have been regenerated.